apps/hash-ai-worker-ts/src/activities/flow-activities/research-entities-action/link-follower-agent/llama-index/simple-storage-context.ts

```diff
@@ -1,98 +1,67 @@
-import { access, mkdir } from "node:fs/promises";
+import fs from "node:fs/promises";
 import path from "node:path";
 import { fileURLToPath } from "node:url";
 
-import type { Subtype } from "@local/advanced-types/subtype";
-import type { StorageContext } from "llamaindex";
-import {
-  SimpleDocumentStore,
-  SimpleIndexStore,
-  SimpleVectorStore,
-} from "llamaindex";
-
-export type SimpleStorageContext = Subtype<
-  StorageContext,
-  {
-    docStore: SimpleDocumentStore;
-    indexStore: SimpleIndexStore;
-    vectorStore: SimpleVectorStore;
-  }
->;
+import { type StorageContext, storageContextFromDefaults } from "llamaindex";
 
 const __filename = fileURLToPath(import.meta.url);
 const __dirname = path.dirname(__filename);
 
 const baseFilePath = path.join(__dirname, "/var/tmp_files");
 
-export const generateSimpleStorageContextFilePaths = (params: {
-  hash: string;
-}) => {
-  const { hash } = params;
-
-  const directoryPath = `${baseFilePath}/storage/${hash}`;
+export interface Storage extends StorageContext {
+  directory: string;
+}
 
-  return {
-    directoryPath,
-    vectorStorePath: `${directoryPath}/vector-store.json`,
-    docStorePath: `${directoryPath}/doc-store.json`,
-    indexStorePath: `${directoryPath}/index-store.json`,
-  };
+const directory = ({ hash }: { hash: string }) => {
+  return `${baseFilePath}/storage/${hash}`;
 };
 
-export const retrieveSimpleStorageContext = async (params: {
-  hash: string;
-}): Promise<{ simpleStorageContext: SimpleStorageContext | undefined }> => {
-  const { hash } = params;
-
-  const { directoryPath, vectorStorePath, docStorePath, indexStorePath } =
-    generateSimpleStorageContextFilePaths({ hash });
+export const createStorageContext = async ({ hash }: { hash: string }) => {
+  const directoryPath = directory({ hash });
 
   try {
-    // Check directory exists
-    await access(directoryPath);
-
-    await Promise.all([
-      access(vectorStorePath),
-      // access(docStorePath),
-      access(indexStorePath),
-    ]);
+    await fs.mkdir(directoryPath, { recursive: true });
+  } catch (error: unknown) {
+    if ((error as NodeJS.ErrnoException).code !== "EEXIST") {
+      // eslint-disable-next-line no-console
+      console.info(
```

Comment:
I've added the required `import` above
```suggestion
      logger.info(
```